sql/share/errmsg-utf8.txt

```diff
@@ -10780,3 +10780,5 @@ ER_CM_OPTION_MISSING_REQUIREMENT
         eng "CHANGE MASTER TO option '%s=%s' is missing requirement %s"
 ER_SLAVE_STATEMENT_TIMEOUT 70100
         eng "Slave log event execution was interrupted (slave_max_statement_time exceeded)"
+ER_SECURE_TRANSPORT_REQUIRED 3159
```

Comment:
eh, no. you didn't change the error number, you've made sqlstate to be 3159, which isn't even a valid sqlstate value.

you can see error numbers in the include/mysqld_error.h file, they're generated, error messages are simply numbered sequentially.

you can see that the last MySQL error in errmsg-utf8.txt is currently ER_ALTER_OPERATION_NOT_SUPPORTED_REASON_GIS, number 3060. So if you want to add a new one with the number 3159, you need to add ~100 new error messages. Like this
```diff
         eng "Do not support online operation on table with GIS index"
         spa "No soporta operación en línea en tabla con índice GIS"
+
+ER_MYSQL_3061
+        eng ""
+ER_MYSQL_3062
+        eng ""
... and so on ...
+ER_MYSQL_3157
+        eng ""
+ER_MYSQL_3158
+        eng ""
+
+ER_SECURE_TRANSPORT_REQUIRED 08004
+        eng "Connections using insecure transport are prohibited while --requi>
+
```

Reply:
> let's try to have it the same number as in MySQL, if possible

I completely understand the goal here, but how is this going to be possible in the long run?

If the wire protocol associates error codes with specific meanings purely based on their numeric codes, which are sequentially increasing numbers assigned by two diverging codebases (MySQL and MariaDB), then there's inevitably going to be a point where MySQL and MariaDB assign different meanings to the same code, right?

(Maybe it has happened already :man_shrugging:)

> So if you want to add a new one with the number 3159, you need to add ~100 new error messages.

Couldn't we simply skip to a new position using `start-error-number`, as done at several other points in the file? E.g.

```diff
 ER_ALTER_OPERATION_NOT_SUPPORTED_REASON_GIS
         chi "不要支持使用GIS索引的表中的在线操作"
         eng "Do not support online operation on table with GIS index"
         spa "No soporta operación en línea en tabla con índice GIS"
+
+ start-error-number 3159
+ ER_SECURE_TRANSPORT_REQUIRED 08004
+        eng "Connections using insecure transport are prohibited while --requi>
+
```

Reply:
1. Ranges are different. 1000-1999 is the historical range for MySQL/MariaDB common errors, 2000-2999 is for client-side errors from Connector/C, 3000-3999 is the new MySQL error range, 4000-4999 is the new MariaDB error range.
2. no we cannot, I even tried that before writing a previous comment. it is assumed that ranges are all 1000 messages wide, see how `ER()` macro is defined:
```c
#define ERRORS_PER_RANGE 1000

#define ER_THD(thd,X) ((thd)->variables.errmsgs[((X)-ER_ERROR_FIRST) / ERRORS_PER_RANGE][(X) % ERRORS_PER_RANGE])
#define ER(X)         ER_THD(current_thd, (X))
```